ThirdParty/netif/gmlc/netif/NetIF.hpp

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright © 2019,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance
+for Sustainable Energy, LLC.  See the top-level NOTICE for additional details.
+All rights reserved. SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <stdio.h>
+#include <stdlib.h>
+
+// other platforms might need different headers -- detecting with macros:
+// https://sourceforge.net/p/predef/wiki/OperatingSystems/
+// https://stackoverflow.com/questions/142508/how-do-i-check-os-with-a-preprocessor-directive
+#if defined(_WIN32)
+  #include <winsock2.h>
+  #include <WS2tcpip.h>
+  #include <iphlpapi.h>
+
+  #pragma comment(lib, "IPHLPAPI.lib")
+  #pragma comment(lib, "Ws2_32.lib")
+#else
+  #include <arpa/inet.h>
+  #include <ifaddrs.h>
+#endif
+
+namespace gmlc
+{
+namespace netif
+{
+  inline std::string addressToString(struct sockaddr *addr)
```

Comment:
we probably should add a doxygen comment on these functions stating what they do